components/Providers/ModalProvider.tsx

```diff
@@ -0,0 +1,24 @@
+import { createContext, ReactNode, useState } from 'react';
+
+export type Modal = {
+  state: 'view' | 'edit' | 'add';
+  type: 'cat' | 'zone';
+  back?: Modal | undefined;
+};
+
+export interface ContextModal {
+  modal: Modal | undefined;
+  setModal: (modal: Modal | undefined) => void;
```

Comment:
Please look for a more explicit name, `Modal` is too generic to be helpfull (ex: `modalConfig`, `setModalConfig`)